iothub/device/src/Common/Extensions/CommonExtensions.cs

```diff
@@ -202,5 +202,16 @@ public static int NthIndexOf(this string str, char value, int startIndex, int n)
 
             return entryIndex;
         }
+
+        /// <summary>
+        /// Helper to remove extra whitespace from the supplied string.
+        /// It makes sure that strings that contain space characters are preserved, and all other space characters are discarded.
```

Comment:
I'm confused. Elsewhere we say we'll remove the white space, but this says we'll preserve them?

Reply:
whitespace within a sentence is preserved

Reply:
I'll update the comment to call it out as a statement (and not a string)